### PR TITLE
Extract static data

### PR DESCRIPTION
**Pull request by DanRStevens**
This is some refactoring to support issue #6.

This extracts hardcoded values into data. It is a simple refactor, with no change in behaviour.

### CODE REVIEW

File: MissionTable.cpp [hunk 1]
```diff
@@ -26,6 +26,15 @@ void WriteBoolCell(bool boolean, std::streamsize cellWidthInChars);
 
 std::string ConvertMissionTypeToString(MissionTypes missionType);
 
+constexpr std::array<std::string_view, 7> columnTitles{
```
Comment (Brett208) [inline MissionTable.cpp:30]: I wouldn't have thought to use string_view in the context of storing in a array. I wonder how that stores the underlying data, as a char* or as a std::string?

File: MissionTable.cpp [hunk 2]
```diff
@@ -115,32 +134,16 @@ void WriteBoolCell(bool boolean, std::streamsize cellWidthInChars)
...
-		if (static_cast<int>(missionType) > 0) {
-			return "Campaign";
-		}
-
-		throw std::runtime_error("An improper MissionType enum value of " + std::to_string(missionType) + " was provided.");
+	// Positive missionTypes represent campaign mission index.
+	if (static_cast<int>(missionType) > 0) {
+		return "Campaign";
 	}
+	auto missionTypeIndex = static_cast<std::size_t>(-missionType) - 1;
```
Comment (Brett208) [inline MissionTable.cpp:144]: Probably worth a comment here explaining why missionType is being multiplied by negative one just to be cast to a size_t.

**Review by Brett208 — APPROVED** [review]
Requested a code comment on one line. Otherwise, looks like good changes. Thanks for working on this.

File: MissionTable.cpp (hunk 1 above)
Comment (DanRStevens) [reply MissionTable.cpp:30]: The raw `const char*` null terminated C-strings from the literals are stored in the global or read-only section of the executable. Either way, their lifetime is the lifetime of the program. The `std::string_view` instances in the array are then references to the the `const char*` literals along with their length. The terminating null is irrelevant to them at that point.

File: MissionTable.cpp (hunk 2 above)
Comment (DanRStevens) [reply MissionTable.cpp:144]: Good point.

**Pull request merged**